pkg/backend/httpstate/client/client.go

```diff
@@ -1370,6 +1370,58 @@ func (pc *Client) SubmitAIPrompt(ctx context.Context, requestBody interface{}) (
 	return res, err
 }
 
+func getCopilotAPIPath() string {
+	return "/api/ai/chat/preview"
+}
+
+// SummarizeErrorWithCopilot summarizes Pulumi Update output using the Copilot API
+func (pc *Client) SummarizeErrorWithCopilot(
+	ctx context.Context,
+	orgID string,
+	lines []string,
+	model string,
+	maxSummaryLen int,
+) (string, error) {
+	request := createSummarizeUpdateRequest(lines, orgID, model, maxSummaryLen)
+
+	jsonData, err := json.Marshal(request)
+	if err != nil {
+		return "", fmt.Errorf("preparing request: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, pc.apiURL+getCopilotAPIPath(), bytes.NewBuffer(jsonData))
+	if err != nil {
+		return "", fmt.Errorf("creating request: %w", err)
+	}
+
+	req.Header.Set("X-Pulumi-Origin", "api.pulumi.com")
```

Comment:
We should remove this, it is only needed for local testing or when targeting 'app'
We should add X-Pulumi-Source="Pulumi CLI"

Reply:
Done